Kernel/Arch/x86/ISABus/I8042Controller.cpp

```diff
@@ -111,16 +111,21 @@ UNMAP_AFTER_INIT ErrorOr<void> I8042Controller::detect_devices()
         configuration &= ~I8042ConfigurationFlag::FirstPS2PortInterrupt;
         configuration &= ~I8042ConfigurationFlag::SecondPS2PortInterrupt;
 
+        // Note: Drain output buffer to ensure we are in a known state before setting
+        // the scan code.
+        TRY(drain_output_buffer());
```

Comment:
I don't think this is safe if user is mashing keyboard keys/moving the mouse (as people sometimes do at PC startup). Maybe a better approach would be to consume any garbage as part of the expected output? Even if they aren't mashing the keys there's still a race window here.